<a href="https://colab.research.google.com/github/Lexie-D/SAMSum-BERT-Dialogue-Summarization/blob/main/Samsum_dialogue_summarization.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Load the SAMSum Dataset

In [22]:
#!pip install datasets
"""
from datasets import load_dataset
from google.colab import files
import shutil

# 1. Load the SAMSum dataset from Hugging Face
print("📥 Downloading SAMSum dataset...")
dataset = load_dataset("knkarthick/samsum")

# 2. Save it to disk in Colab
dataset.save_to_disk("samsum_dataset")
print("💾 Dataset saved to folder 'samsum_dataset'.")

# 3. Zip the folder
shutil.make_archive("samsum_dataset", 'zip', "samsum_dataset")
print("📦 Dataset zipped as 'samsum_dataset.zip'.")

# 4. Download to your computer
files.download("samsum_dataset.zip")
print("✅ Download started — check your Downloads folder.")
"""

📥 Downloading SAMSum dataset...


Saving the dataset (0/1 shards):   0%|          | 0/14732 [00:00<?, ? examples/s]

Saving the dataset (0/1 shards):   0%|          | 0/818 [00:00<?, ? examples/s]

Saving the dataset (0/1 shards):   0%|          | 0/819 [00:00<?, ? examples/s]

💾 Dataset saved to folder 'samsum_dataset'.
📦 Dataset zipped as 'samsum_dataset.zip'.


<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

✅ Download started — check your Downloads folder.


In [24]:
from google.colab import files
import zipfile
from datasets import load_from_disk

# Upload from local computer
uploaded = files.upload()

# Unzip the uploaded file
with zipfile.ZipFile("samsum_dataset.zip", 'r') as zip_ref:
    zip_ref.extractall("samsum_dataset")

# Load dataset
dataset = load_from_disk("samsum_dataset")

print(dataset)
print(dataset["train"][0])

Saving samsum_dataset.zip to samsum_dataset (1).zip
DatasetDict({
    train: Dataset({
        features: ['id', 'dialogue', 'summary'],
        num_rows: 14732
    })
    validation: Dataset({
        features: ['id', 'dialogue', 'summary'],
        num_rows: 818
    })
    test: Dataset({
        features: ['id', 'dialogue', 'summary'],
        num_rows: 819
    })
})
{'id': '13818513', 'dialogue': "Amanda: I baked  cookies. Do you want some?\nJerry: Sure!\nAmanda: I'll bring you tomorrow :-)", 'summary': 'Amanda baked cookies and will bring Jerry some tomorrow.'}


### Split is Train(14,732) / Validation(818) / Test(819)
### Each Row has...
- ID: unique identifier
- Dialogue: chat conversation
- Summary: human written summary

## Access Data

In [12]:
#Example dialogue train:0
print(dataset["train"][0]["dialogue"])

#example summary train:0
print(dataset["train"][0]["summary"])

Amanda: I baked  cookies. Do you want some?
Jerry: Sure!
Amanda: I'll bring you tomorrow :-)
Amanda baked cookies and will bring Jerry some tomorrow.


In [28]:
!find /content -name "*.ipynb"

In [25]:
%cd /content/SAMSum-BERT-Dialogue-Summarization

/content/SAMSum-BERT-Dialogue-Summarization


In [26]:
!git clone https://github.com/Lexie-D/SAMSum-BERT-Dialogue-Summarization.git
%cd SAMSum-BERT-Dialogue-Summarization

fatal: destination path 'SAMSum-BERT-Dialogue-Summarization' already exists and is not an empty directory.
/content/SAMSum-BERT-Dialogue-Summarization/SAMSum-BERT-Dialogue-Summarization


In [27]:
!mv /content/YourNotebookName.ipynb .

mv: cannot stat '/content/YourNotebookName.ipynb': No such file or directory


In [29]:
!find /content -name "*.ipynb"

In [ ]:
import getpass
token = getpass.getpass('Enter your GitHub token: ')

!git add .
!git commit -m "Added SAMSum dataset loading code"
!git push https://Lexie-D:{token}@github.com/Lexie-D/SAMSum-BERT-Dialogue-Summarization.git
